src/main/java/hudson/plugins/warnings/WarningsBuildHistory.java

```diff
@@ -27,15 +28,38 @@
      * @param useStableBuildAsReference
      *            determines whether only stable builds should be used as
      *            reference builds or not
+     *
+     * @deprecated use {@link #WarningsBuildHistory(Run, String, boolean, boolean)}
      */
+    @Deprecated
     public WarningsBuildHistory(final AbstractBuild<?, ?> lastFinishedBuild, @CheckForNull final String group,
             final boolean usePreviousBuildAsReference, final boolean useStableBuildAsReference) {
+        this((Run<?, ?>) lastFinishedBuild, group, usePreviousBuildAsReference, useStableBuildAsReference);
+    }
+
+    /**
+     * Creates a new instance of {@link WarningsBuildHistory}.
+     *
+     * @param lastFinishedBuild
+     *            the last finished build
+     * @param group
+     *            the parser group
+     * @param usePreviousBuildAsReference
+     *            determines whether to use the previous build as the reference
+     *            build
+     * @param useStableBuildAsReference
+     *            determines whether only stable builds should be used as
+     *            reference builds or not
+     */
+    public WarningsBuildHistory(final Run<?, ?> lastFinishedBuild, @CheckForNull final String group,
+                                final boolean usePreviousBuildAsReference, final boolean useStableBuildAsReference) {
         super(lastFinishedBuild, WarningsResultAction.class, usePreviousBuildAsReference, useStableBuildAsReference);
 
         this.group = group;
     }
 
     @Override
+    @Deprecated
```

Comment:
NIT: you may want to add a comment. Almost everywhere in the code you put a comment before the deprecated function explaining which one to use instead, but here and somewhere else in the code you did not.